src/compatibility/fvSolution.cpp

```diff
@@ -105,6 +105,15 @@ void updatePreconditioner(NeoN::Dictionary& solverDict)
 
 void updateCriteria(NeoN::Dictionary& solverDict)
 {
+    // parse given dictionary, get numeric value of key in a safe way
+    auto safeRetrieveScalarAndRemove = [](NeoN::Dictionary& d, std::string key)
```

Comment:
The function is a little too long.

```suggestion
    auto extractScalar = [](NeoN::Dictionary& d, std::string key)
```
"Extract" implies _retrieve + remove_